hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContractRevision.php

```diff
@@ -147,5 +118,54 @@ static function importableFields($contactType = 'HRJobContractRevision',
     }
     return $fields;
   }
-  
+
+  /**
+   * Helper function used in updateEffectiveEndDates() loop.
+   * It updates revision entry with calculated 'effective_end_date'
+   * and 'overrided' values basing on specified date of previously iterated
+   * revision and date of currenly iterated revision.
+   * 
+   * @param int $revisionId
+   * @param string $previousEffectiveDate
+   * @param string $currentEffectiveDate
+   */
+  protected static function _updateRevisionByEffectiveDates($revisionId, $previousEffectiveDate, $currentEffectiveDate) {
+    $overrided = 0;
+    $effectiveEndDate = $currentEffectiveDate;
+    if ($previousEffectiveDate === $currentEffectiveDate) {
+      // If 'effective_date' is equal to previously iterated revision's
+      // effective date we mark this revision as overrided by the one newly
+      // created.
+      $overrided = 1;
+    } else {
+      // Else, we set current's revision effective day to a day before the
+      // previously iterated (newer) one.
+      $effectiveEndDate = date('Y-m-d', strtotime($previousEffectiveDate) - 3600 * 24);
```

Comment:
instead of this seconds * hours math, what about this?:
`$effectiveEndDate = (new Datetime($previousEffectiveDate))->modify('+1 day')`